src/ngraph/runtime/cpu/kernel_selectors.hpp

```diff
@@ -0,0 +1,305 @@
+    else if (ET == element::i32)                                                                   \
+    {                                                                                              \
+        S(K, KV, ##__VA_ARGS__, int32_t);                                                          \
+    }                                                                                              \
+    else if (ET == element::i64)                                                                   \
+    {                                                                                              \
+        S(K, KV, ##__VA_ARGS__, int64_t);                                                          \
+    }                                                                                              \
+    else if (ET == element::u8)                                                                    \
+    {                                                                                              \
+        S(K, KV, ##__VA_ARGS__, uint8_t);                                                          \
+    }                                                                                              \
+    else if (ET == element::u16)                                                                   \
+    {                                                                                              \
+        S(K, KV, ##__VA_ARGS__, uint16_t);                                                         \
+    }                                                                                              \
+    else if (ET == element::u32)                                                                   \
+    {                                                                                              \
+        S(K, KV, ##__VA_ARGS__, uint32_t);                                                         \
+    }                                                                                              \
+    else if (ET == element::u64)                                                                   \
+    {                                                                                              \
+        S(K, KV, ##__VA_ARGS__, uint64_t);                                                         \
+    }                                                                                              \
+    else                                                                                           \
+        throw ngraph_error("Unsupported element type " + ET.c_type_string() + " for kernel " #K);
+
+// Expand only selected datatypes. Named macros (e.g., F32_SELECT) are expanded based on build-flags
+#define EXPAND_ETS(K, KV, ET, S, ...)                                                              \
+    if (ET == element::boolean)                                                                    \
```

Comment:
Shouldn't this be `if (BOOLEAN_EN && ET == element::boolean)` ? Otherwise we can have a segfault at runtime instead of going to the ngraph_error defined below. Same for all following if cases.